test/boost/aws_errors_test.cc

```diff
@@ -0,0 +1,445 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#define BOOST_TEST_MODULE object_storage
+
+#include <boost/test/unit_test.hpp>
+#include <seastar/core/sstring.hh>
+#include <utils/s3/aws_errors.hh>
+
+uint8_t singular = 1;
+uint8_t plural = 2;
+
+namespace boost::test_tools::tt_detail {
+template <>
+struct print_log_value<aws::aws_error_type> {
+    void operator()(std::ostream& os, const aws::aws_error_type& error_type) { os << static_cast<uint8_t>(error_type); }
```

Comment:
instead of specializing in 

i'd suggest adding `boost_test_print_type()` in our own namespace, like:
```c++
namespace aws {

std::ostream& boost_test_print_type(std::ostream& os,  const aws::aws_error_type& error_type) {
    return os << fmt::underlying(error_type);
}

}
```

or use `format_as()`, if you can afford including `lib/test_utils.hh`.

Reply:
done